item.rb

```diff
@@ -0,0 +1,36 @@
+require 'date'
+
+class Item
+  attr_reader :id, :archived
+  attr_accessor :publish_date
+
+  def initialize(id, publish_date)
+    @id = rand(1..1000)
```

Comment:
It would be better if we can generate a random id only if the user doesn't enter one, this line can be something like : `id = id || rand(1..100_000)`